pkg/provider/azure_loadbalancer.go

```diff
@@ -2179,7 +2179,7 @@ func (az *Cloud) reconcileSecurityGroup(clusterName string, service *v1.Service,
 
 	var sourceAddressPrefixes []string
 	if (sourceRanges == nil || servicehelpers.IsAllowAll(sourceRanges)) && len(serviceTags) == 0 {
-		if !requiresInternalLoadBalancer(service) {
+		if !requiresInternalLoadBalancer(service) || len(service.Spec.LoadBalancerSourceRanges) > 0 {
 			sourceAddressPrefixes = []string{"Internet"}
```

Comment:
If reaching this line, the `servicehelpers.IsAllowAll(sourceRanges))` needs to be true, which means the source range will include 0.0.0.0/0. In this case, the user allows the traffic from outside. LGTM.